app/controllers/users_controller.rb

```diff
@@ -37,17 +37,60 @@ def change_password
     render json: message, status: :ok
   end
 
+  def do_transaction
+    sender = User.find(transaction_params[:sender_id])
+    receiver = User.find(transaction_params[:receiver_id])
+    amount = transaction_params[:amount]
+    entered_pin = transaction_params[:entered_pin].to_s
+
+    if sender.id == receiver.id
+      return render json: { errors: "Sender and receiver cannot be the same" },
+      status: :unprocessable_entity
+    end
+
+    if sender.status != "active" || receiver.status != "active"
+      return render json: { errors: "Both accounts must be active" },
```

Comment:
same process has been followed in each if block, can we move it into one method.